azblob/zt_client_provided_key_test.go

```diff
@@ -0,0 +1,504 @@
+package azblob
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/base64"
+	"encoding/binary"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	chk "gopkg.in/check.v1" // go get gopkg.in/check.v1
+)
+
+/*
+Azure Storage supports following operations support of sending customer-provided encryption keys on a request:
+Put Blob, Put Block List, Put Block, Put Block from URL, Put Page, Put Page from URL, Append Block,
+Set Blob Properties, Set Blob Metadata, Get Blob, Get Blob Properties, Get Blob Metadata, Snapshot Blob.
+*/
+var testEncryptedKey = "MDEyMzQ1NjcwMTIzNDU2NzAxMjM0NTY3MDEyMzQ1Njc="
+var testEncryptedHash = "3QFFFpRA5+XANHqwwbT4yXDmrT/2JaLt/FKHjzhOdoE="
+var testEncryptedScope = ""
```

Comment:
>testEncryptedScope [](start = 4, length = 18)

It's hard to set up tests for CPK-N in our unit tests (we probably need keyvault), but maybe you could do a one-off test? Just to make sure that CPK-N works just as well as CPK-V

Reply:
Yeah I was thinking about it for sometime. I wonder why python SDK didn't have any such tests. Let's discuss this is the sync-up

Reply:
Done